test/suite_test.go

```diff
@@ -105,11 +107,21 @@ func (e *kinde2e) SetupSuite() {
 	e.Nil(os.MkdirAll(buildDir, 0o755))
 
 	e.kindPath = filepath.Join(buildDir, "kind")
-	e.downloadAndVerify(
-		"https://github.com/kubernetes-sigs/kind/releases/download/"+
-			kindVersion+"/kind-linux-amd64",
-		e.kindPath, kindSHA512,
-	)
+	operatingSystem := runtime.GOOS
+	switch operatingSystem {
+	case "darwin":
+		e.downloadAndVerify(
+			"https://github.com/kubernetes-sigs/kind/releases/download/"+
+				kindVersion+"/kind-darwin-amd64",
+			e.kindPath, kindOSXSHA512,
+		)
```

Comment:
Let's create a new variable for the SHA (based on the operating system) and factor out the string building to avoid the code duplication.

Reply:
It is just not SHA. It is also the platform `kind-darwin-amd64` and `kind-linux`. This will end up with a bunch of string formatting.  

I agree code duplication is not encouraged, but this helps in readability. Thoughts?

Reply:
Took care of it. Thanks